### PR TITLE
[13.0][ADD] product_expiry_configurable

### PR DESCRIPTION
**Pull request by albariera**
This model allows setting expiry times on category and to use the 'end_of_life' date for the computation of lot dates.
Read usage for more information.

### CODE REVIEW

**Comment by albariera** [comment]
@LoisRForgeFlow @etobella

**Comment by albariera** [comment]
@lmignon I have used in this module the same inheritance method that you have used in #670. It was very useful thank you! I hope you like this new addon.

**Comment by lmignon** [comment]
> @lmignon I have used in this module the same inheritance method that you have used in #670. It was very useful thank you! I hope you like this new addon.

@albariera Happy it help you.

Thank you for your proposal, nevertheless tests are missing. Can you add tests for this new addon PLZ?

**Comment by albariera** [comment]
@lmignon Done!! :blush:

**Review by etobella — APPROVED** [review]

File: product_expiry_configurable/models/stock_production_lot.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import datetime
+
+from odoo import api, fields, models
+
+
+class StockProductionLot(models.Model):
+
+    _inherit = "stock.production.lot"
+
+    product_qty = fields.Float(store=True)
```
Comment (etobella) [inline product_expiry_configurable/models/stock_production_lot.py]: Why do you need this stored?

Comment (albariera) [reply product_expiry_configurable/models/stock_production_lot.py]: I was searching among lots using this field, but it is true that it does not have much sense. I remove it. Thank you!

**Comment by rousseldenis** [comment]
@albariera Thanks for this. I would say the module name should be a little bit improved as 'configurable' is a little bit too generic (but I don't have a specific idea - as your module does a lot of things...)

**Comment by albariera** [comment]
@rousseldenis I agree with you that the name is too generic, but I could not come up with a better name. Maybe product_expiry_dates_extended could be better? Do you have any proposals?

File: product_expiry_configurable/models/stock_production_lot.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        }
+        product = self.env["product.product"].browse(product_id) or self.product_id
+        res = dict.fromkeys(mapped_fields, False)
+        life_date = self.life_date or self.env.context.get("life_date")
+        if product and product.tracking != "none":
+            if product.compute_dates_from == "life_date" and life_date:
+                mapped_fields.pop("life_date", None)
+                res = self._get_dates_from_life_date(product, mapped_fields, life_date)
+            elif product.compute_dates_from == "current_date":
+                res = self._get_dates_from_current_date(product, mapped_fields)
```
Comment (LoisRForgeFlow) [inline product_expiry_configurable/models/stock_production_lot.py]: You should use super here

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

File: product_expiry_configurable/models/stock_production_lot.py (hunk 2 above)
Comment (albariera) [reply product_expiry_configurable/models/stock_production_lot.py]: Done!! Thank you for your review :)

**Comment by etobella** [comment]
@OCA/product-maintainers Can you proceed with merge?

**Comment by olgamarcocb** [comment]
LGTM :+1:

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**